src/mlir/generator.cc

```diff
@@ -426,9 +426,40 @@ namespace mlir::verona
     auto expr = parseNode(getSingleSubNode(ast).lock());
     if (auto err = expr.takeError())
       return std::move(err);
+
+    // Check which type of region we're in
     auto region = builder.getInsertionBlock()->getParent();
+    mlir::Type retTy;
     auto op = region->getParentOp();
-    if (isa<mlir::verona::WhileOp>(op))
+    bool isInLoop = isa<mlir::verona::WhileOp>(op);
```

Comment:
Can this variable binding be removed? I don't think it's used until the if on line 462

Reply:
It cannot because the `op` changes if the ast node is deeply nested.

If I let the `isa` check happen after the `while` loop, it will always return `false` when nested.